### PR TITLE
Started adding test cases

### PR DESCRIPTION
**Pull request by codechaitu**
`Why this request?`
To add test cases to application, so that it will be a check.

### CODE REVIEW

File: models/models.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ type DBModel struct {
 	DeletedAt *time.Time `json:"-"`
 }
 
+type LoginValues struct{
```
Comment (singh-shubham) [inline models/models.go]: Do we really need this struct out of testing scope?

File: server/handlers.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func getUserFirstName(c *gin.Context) {
 	id := claims["id"]
 	var user models.User
 	db.Where("id = ?", id).First(&user)
-	c.JSON(200, user.FName)
+	c.String(200, user.FName)
```
Comment (singh-shubham) [inline server/handlers.go]: Revert it back to JSON.
If there is some problem in testing then, typecast the JSON to string during test assertion

File: server/handlers.go [hunk 3]
```diff
@@ -142,7 +142,7 @@ func regParkingSpot(c *gin.Context) {
 		db.Save(&owner)
 	}
 
-	c.JSON(201, "Listed a new parking Spot Successfully!")
+	c.String(201, "Listed a new parking Spot Successfully!")
```
Comment (singh-shubham) [inline server/handlers.go]: See above comment

File: models/models.go (hunk 1 above)
Comment (codechaitu) [reply models/models.go]: In the initial testing phase, we were supposed to give login values to the request, as per that time, I was trying out to give values initialize from a struct. Ya, we can remove that now, as we are embedding the values for test directly into the request. Amended in the commit bf8720051cec65504a9d5c5c108cf4b11a42fbb0

File: server/handlers.go (hunk 2 above)
Comment (codechaitu) [reply server/handlers.go]: Updated in the commit 46c5e524ee9090fa246d58ea5922440dd6f7147c

File: server/handlers.go (hunk 3 above)
Comment (codechaitu) [reply server/handlers.go]: Updated in the commit 46c5e524ee9090fa246d58ea5922440dd6f7147c

File: server/server_test.go [hunk 4]
```diff
@@ -72,7 +84,109 @@ var _ = Describe("Server", func() {
...
+			BeforeEach(func() {
+				body := new(bytes.Buffer)
+				response = performRequest(router, "GET", "/dashboard/1/mylistings", body)
+				fmt.Println((response.Body.String()))
+				//bytes.NewBuffer([]byte(body))
+			})
+
+			It("Returns with Status 200", func() {
+				Expect(response.Code).To(Equal(200))
+			})
```
Comment (singh-shubham) [inline server/server_test.go:134]: Can't we assert against the data that we received? I think that will be a better test than just checking the status!

File: server/server_test.go [hunk 5]
```diff
@@ -72,7 +84,109 @@ var _ = Describe("Server", func() {
...
+									"EndTime" : "2006-01-02T15:00:00.000Z"
+									}`))
+
+				response = performRequest(router, "POST", "/dashboard/1/parkmenow", body)
+				fmt.Println((response.Body.String()))
+			})
+
+			It("Returns with Status 200", func() {
+				Expect(response.Code).To(Equal(200))
+			})
```
Comment (singh-shubham) [inline server/server_test.go:155]: Same, write a test that assert with the response. Not just the response code.

File: server/server_test.go [hunk 6]
```diff
@@ -72,7 +84,109 @@ var _ = Describe("Server", func() {
...
+				fmt.Println((response.Body.String()))
+			})
+
+			It("Returns with Status 201", func() {
+				Expect(response.Code).To(Equal(201))
+			})
+
+			It("Returns saying succesful booking", func() {
+				Expect(response.Body.String()).To(Equal("\"Listed a new parking Spot Successfully!\""))
+			})
```
Comment (singh-shubham) [inline server/server_test.go:182]: Check in the DB weather the Row actually got created or not!

**Review by singh-shubham — APPROVED** [review]

**Pull request merged**